src/qt/optionsmodel.cpp

```diff
@@ -110,6 +110,11 @@ void OptionsModel::Init(bool resetSettings)
         settings.setValue("bSpendZeroConfChange", true);
     if (!gArgs.SoftSetBoolArg("-spendzeroconfchange", settings.value("bSpendZeroConfChange").toBool()))
         addOverriddenOption("-spendzeroconfchange");
+
+    if (!settings.contains("strAddressType"))
+      settings.setValue("strAddressType", QString::fromStdString(FormatOutputType(OUTPUT_TYPE_DEFAULT)));
+    if (!gArgs.SoftSetArg("-addresstype", settings.value("strAddressType").toString().toStdString()))
+        addOverriddenOption("-addresstype");
```

Comment:
Actually this is not true. It will happily ignore `-addresstype` if you change the setting.

Instead, you should add `addOverriddenOption("-changetype");`, since this setting _is_ honored.